systemtest/src/test/java/io/strimzi/systemtest/log/LogSettingST.java

```diff
@@ -412,6 +417,39 @@ void testBridgeLogSetting(ExtensionContext extensionContext) {
         checkContainersHaveProcessOneAsTini(bridgeName);
     }
 
+    @ParallelTest
+    // This test might be flaky, as it gets real logs from CruiseControl pod
+    void testCruiseControlLogChange() {
+        final String debugText = " DEBUG ";
+        String cruiseControlPodName = PodUtils.getPodNameByPrefix(LOG_SETTING_CLUSTER_NAME + "-" + Constants.CRUISE_CONTROL_CONTAINER_NAME);
+        LOGGER.info("Check that default/actual root logging level is info");
+        String containerLogLevel = cmdKubeClient().execInPod(cruiseControlPodName, "grep", "-i", "rootlogger.level",
+                Constants.CRUISE_CONTROL_LOG_FILE_PATH).out().trim().split("=")[1];
+        assertThat(containerLogLevel.toUpperCase(Locale.ENGLISH), is(not(debugText.strip())));
+
+        LOGGER.info("Check logs in CruiseControl - make sure no DEBUG is found there.");
+        String logOut = cmdKubeClient().namespace(cluster.getNamespace()).execInCurrentNamespace("logs", cruiseControlPodName, "-c", Constants.CRUISE_CONTROL_CONTAINER_NAME, "--since=20s").out();
+        assertThat(logOut, not(containsString(debugText)));
+
+        InlineLogging logging = new InlineLogging();
+        logging.setLoggers(Collections.singletonMap("rootLogger.level", debugText.strip()));
+        KafkaResource.replaceKafkaResource(LOG_SETTING_CLUSTER_NAME, kafka -> kafka.getSpec().getCruiseControl().setLogging(logging));
+
+        try {
+            Thread.sleep(CO_OPERATION_TIMEOUT_SHORT);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        LOGGER.info("Verify changed root logger in {}.", cruiseControlPodName);
+        containerLogLevel = cmdKubeClient().execInPod(cruiseControlPodName, "grep", "-i", "rootlogger.level", Constants.CRUISE_CONTROL_LOG_FILE_PATH).out().trim().split("=")[1];
+        assertThat(containerLogLevel.toUpperCase(Locale.ENGLISH), is(debugText.strip()));
+
+        LOGGER.info("Check cruise control logs in pod {} and it's container {} .", cruiseControlPodName, Constants.CRUISE_CONTROL_CONTAINER_NAME);
+        logOut = cmdKubeClient().namespace(cluster.getNamespace()).execInCurrentNamespace("logs", cruiseControlPodName, "-c", Constants.CRUISE_CONTROL_CONTAINER_NAME, "--since=20s").out();
+        assertThat(logOut, containsString(debugText));
```

Comment:
Well, when you want to check the log, I would use the `assumeThat` or `assumeTrue`, and then do the assertion -> in case that the log will not contain the `DEBUG` text, it will end without error - we don't want flaky tests, right?

Reply:
Oh really? I thought it would fail with an error if debugText is not in logOut. I will check it out as well.
Thanks.

Reply:
Well I would at least try it, thanks :)

Reply:
@im-konge What is the point of the assert if you skip it  instead of letting it fail?

Reply:
We discussed offline that checking log is not good and we're trying to avoid it. I guess it's better (if this assert is only optional and we're checking the log change in the log4j file) to do the `assumeTrue` and then do some checks or remove the assert, because having flaky test is not wanted from my POV.

Reply:
Or add some `waitFor` for this

Reply:
So I read what `assumeThat|True` does and I think it does not think it fits well into this test. I am going with waitFor usage instead. ;-)

Reply:
@im-konge The thing is that if it is optional it doesn't really matter, or? I can understand that the tests with real logs can be flaky. However:
* What value does it have to check it if you ignore the result? You can probably just remove it in that case and save time at least, or?
* Without checking the actual logs - do we really need STs? Updating the ConfigMap resource can be quite easily tested in UTs. Updating the file in the Pod is probably tested quite well in Kube.

Reply:
Yeah I know, that without the log check it's useless to have these kind of tests. All I don't want is to have a flaky test. So if the `waitFor` will "fix" it, it's good. Thanks @scholzj and @michalxo